### PR TITLE
chore: remove FEATURE_BEAT flag

### PR DESCRIPTION
**Pull request by wescopeland**
This is a housekeeping PR. 

Now that progression features are enabled, we no longer need all the feature flag conditionals for `FEATURE_BEAT`. This PR removes all of them.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -15,7 +15,8 @@ FORWARD_PHPMYADMIN_PORT=64080
 
 # Feature Flags
 
-FEATURE_BEAT=true
+## example
+#FEATURE_BEAT=true
```
Comment (luchaos) [inline .env.example:19]: `#FEATURE_EXAMPLE=true`

File: config/feature.php [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
-    /*
-     * If true, player-facing beaten games UI/UX is enabled.
-     */
-    'beat' => env('FEATURE_BEAT', false),
-
+/**
+ * EXAMPLE:
+ * // If true, player-facing beaten games UI/UX is enabled.
+ * 'beat' => env('FEATURE_BEAT', false),
+ */
```
Comment (luchaos) [inline config/feature.php:8]: // 'example' => env('FEATURE_EXAMPLE', false),

**Review by luchaos — APPROVED** [review]

File: .env.example (hunk 1 above)
Comment (Jamiras) [inline .env.example:19]: There's also a reference in `.env.testing`

Comment (wescopeland) [reply .env.example]: Resolved in https://github.com/RetroAchievements/RAWeb/pull/2059/commits/e57d45732f1ccd3c1694bb9eb63b62d26222a0f6.

See also: https://github.com/RetroAchievements/RAWeb/pull/2059#discussion_r1405221043

File: config/feature.php (hunk 2 above)
Comment (wescopeland) [reply config/feature.php]: Resolved in https://github.com/RetroAchievements/RAWeb/commit/e57d45732f1ccd3c1694bb9eb63b62d26222a0f6.

File: .env.example (hunk 1 above)
Comment (wescopeland) [reply .env.example]: Resolved in https://github.com/RetroAchievements/RAWeb/commit/e57d45732f1ccd3c1694bb9eb63b62d26222a0f6.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**